include/gz/rendering/Material.hh

```diff
@@ -220,8 +221,31 @@ namespace gz
       public: virtual std::string Texture() const = 0;
 
       /// \brief Set the material texture
-      /// \param[in] _texture URI of the new texture file
-      public: virtual void SetTexture(const std::string &_texture) = 0;
+      /// \param[in] _texture URI of the new texture file or identifying name
+      /// if _img is set
+      public: virtual void GZ_DEPRECATED(7) SetTexture(
+        const std::string &_texture)
```

Comment:
just thinking about this, we could offer an API that defaults the `_img` arg to `nullptr` then we don't need to do the deprecation? We'll just have one `SetTexture` function:

```
virtual void SetTexture(const std::string &_texture, const std::shared_ptr<const common::Image> &_img = nullptr)
```

User's old code should still work. This breaks ABI but that should be fine since it's a new version.

Reply:
From what I understand @chapulina 's proposal for the current API [here](https://github.com/gazebosim/gz-rendering/pull/642#discussion_r930175716) was motivated by the fact that if the signature of the function was changed, users who inherit from `Material` and have a 

```
virtual void SetTexture(const std::string &_texture) override;
```

would end up having a compile error without a formal "tick" since their `override` function does not override anymore.
I can see arguments both ways so not sure what's better

Reply:
oh I see, I missed that. Ok looks like we'll need to keep the existing one then. Can its implementation just call 

```
this->SetTexture(_texture, nullptr)
```

Reply:
Good point! I gave it a quick spin in the beginning but I had some trouble with undefined references that made me just create empty functions for all of them.
Done https://github.com/gazebosim/gz-rendering/pull/642/commits/334b28a250ac75f32d03e71586849fcdf3314d94